### PR TITLE
SINGA-474 prelu,add,equal,selu,elu operator

### PR DESCRIPTION
**Pull request by ShichengChen**

### CODE REVIEW

File: python/singa/autograd.py [hunk 1]
```diff
@@ -608,21 +608,177 @@ def backward(self, dy):
...
     def forward(self, a, b):
+        #up till now, the dimensions of tensor a and b should less than 3
+        self.shape0=list(a.shape())
+        self.shape1=list(b.shape())
+        assert(len(self.shape0) <= 2 and len(self.shape1) <= 2),"up till now, the dimensions of tensor a and b should less than 3"
         return singa.__add__(a, b)
 
     def backward(self, dy):
-        return dy, dy
+        if(type(dy)==float):return dy,dy
```
Comment (nudles) [inline python/singa/autograd.py]: this implementation assumes that the a has the same shape as dy?
then need to assert it.

File: python/singa/autograd.py [hunk 2]
```diff
@@ -608,21 +608,177 @@ def backward(self, dy):
...
+            a CTensor for the result
+        """
+        #f(x) = alpha * (exp(x) - 1.) for x < 0, f(x) = x for x >= 0
+        if training:
+            self.input = x
+        x1 = singa.LTFloat(x, 0.0)
+        x1 = singa.__mul__(x, x1)
+        x1 = singa.MultFloat(singa.SubFloat(singa.Exp(x1),1.0),self.alpha)
+        x2 = singa.ReLU(x)
+        x1 = singa.__add__(x1, x2)
```
Comment (nudles) [inline python/singa/autograd.py]: can use ```x1+=x2```?

File: python/singa/autograd.py [hunk 3]
```diff
@@ -608,21 +608,177 @@ def backward(self, dy):
...
+            a tuple for dx
+        """
+        dx1mask = singa.LTFloat(self.input, 0.0)
+        dx1 = singa.MultFloat(singa.Exp(self.input), self.alpha)
+        dx1 = singa.__mul__(dx1mask, dx1)
+
+        dx2mask = singa.GEFloat(self.input, 0.0)
+
+        dx = singa.__add__(dx1, dx2mask)
+        return singa.__mul__(dy, dx)
```
Comment (nudles) [inline python/singa/autograd.py]: can use dx *= dy?

File: test/python/test_operation.py [hunk 4]
```diff
@@ -1587,25 +1587,148 @@ def test_min_gpu(self):
...
+            if(gpu):
+                dy.to_device(gpu_dev)
+                x.to_device(gpu_dev)
+            dx = result.creator.backward(dy.data)
+            np.testing.assert_array_almost_equal(tensor.to_numpy(result), y, decimal=5)
+            np.testing.assert_array_almost_equal(tensor.to_numpy(tensor.from_raw_tensor(dx)), grad, decimal=5)
+        test_helper(False)
+        test_helper(True)
+
+    @unittest.skipIf(not singa_wrap.USE_CUDA, 'CUDA is not enabled')
```
Comment (nudles) [inline test/python/test_operation.py]: will it skip the cpu test as well?

Comment (ShichengChen) [reply test/python/test_operation.py]: I just changed it to if-else checks inside unit test functions

File: python/singa/autograd.py (hunk 1 above)
Comment (ShichengChen) [reply python/singa/autograd.py]: yes

File: python/singa/autograd.py (hunk 2 above)
Comment (ShichengChen) [reply python/singa/autograd.py]: yes

File: python/singa/autograd.py (hunk 3 above)
Comment (ShichengChen) [reply python/singa/autograd.py]: yes

**Pull request merged**